silverback/middlewares.py

```diff
@@ -66,34 +64,40 @@ def fix_dict(data: dict, recurse_count: int = 0) -> dict:
         return message
 
     def _create_label(self, message: TaskiqMessage) -> str:
-        if message.task_name == "block":
-            args = f"[block={message.args[0].hash.hex()}]"
-
-        elif "event" in message.task_name:
-            args = f"[txn={message.args[0].transaction_hash},log_index={message.args[0].log_index}]"
+        if labels_str := (
+            ",".join(f"{k}={v}" for k, v in message.labels.items() if k != "task_name")
+        ):
+            return f"{message.task_name}[{labels_str}]"
 
         else:
-            args = ""
-
-        return f"{message.task_name}{args}"
+            return message.task_name
 
     def pre_execute(self, message: TaskiqMessage) -> TaskiqMessage:
-        if message.task_name == "block":
+        message.labels["task_name"] = message.task_name
+        task_type = message.labels.pop("task_type", "<unknown>")
+
+        # NOTE: Don't compare `str` to `TaskType` using `is`
```

Comment:
this note seems odd, probably could be easier-ly read but adjusting line 77 to be:

```
task_type: str = message.labels.pop("task_type", "<unknown>")
```

(giving it a type declare for readability and then obviously you wouldnt use `is`)

Reply:
this bit me in the ass because I thought I could do a direct compare with a str Enum (I made my own because StrEnum is only 3.11+)

funny thing is the stdlib implementation raises a warning that would have helped:
```py
SyntaxWarning: "is" with a literal. Did you mean "=="?
```

Reply:
https://github.com/ApeWorX/silverback/pull/66/commits/6aafc8326b742637c2a74cbf09aa394895f86717

Reply:
ya i never really like IntEnum or StrEnum for reasons like this, too confusing. what is what anymore.